src/execute_linter.cc

```diff
@@ -13,55 +13,157 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+#include "src/execute_linter.h"
+
+#include <cctype>
 #include <cstdio>
 #include <fstream>
 #include <iostream>
+#include <map>
 #include <memory>
 #include <streambuf>
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_split.h"
 #include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+#include "re2/re2.h"
 #include "src/lint_errors.h"
 #include "src/linter.h"
+#include "src/linter_options.h"
 #include "zetasql/base/status.h"
 #include "zetasql/base/status_macros.h"
 
 namespace zetasql::linter {
 
-namespace {
-// It runs all specified checks
-// "LinterOptions" parameter will be added in future.
-LinterResult RunChecks(absl::string_view sql) {
+// This function will parse "NOLINT"(<CheckName>) syntax from
+// a single line of comment.
+// If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+LinterResult ParseNoLintSingleComment(absl::string_view line,
+                                      const absl::string_view& sql,
+                                      LinterOptions* options, int position) {
   LinterResult result;
-  result.Add(CheckLineLength(sql));
-  result.Add(CheckParserSucceeds(sql));
-  result.Add(CheckSemicolon(sql));
-  result.Add(CheckUppercaseKeywords(sql));
-  result.Add(CheckCommentType(sql));
-  result.Add(CheckAliasKeyword(sql));
-  result.Add(CheckTabCharactersUniform(sql));
-  result.Add(CheckNoTabsBesidesIndentations(sql));
+  std::map<std::string, ErrorCode> error_map = GetErrorMap();
+
+  const LazyRE2 kRegex = {"\\s*(NOLINT|LINT)\\s*\\(([a-z ,-]*)\\)\\s*(.*)\\s*"};
+  std::string type = "";
+  std::string check_names = "";
+  std::string lint_comment = "";
+  bool matched =
+      RE2::FullMatch(line, *kRegex, &type, &check_names, &lint_comment);
+
+  // Found enabling or disabling type of comment.
+  if (matched) {
+    check_names.erase(
```

Comment:
I used 'absl::StripAsciiWhitespace' but i think it only erases from middle and end. Then, it fails in the test because of whitespace' ' that comes after comma ' '. I want to erase all whitespaces. This works fine in that sense.